### DISCUSSION TITLE
Accidently Deleted "BlockBrush.dll"

### DISCUSSION THREAD

**Discussion by AshleyPikachu** [PintaProject/Pinta#875] [Q&A] [answered]
Hi, where can I get another one as a reinstall did not fix it. This is for the development build of Pinta.

**Comment by cameronwhite** [comment]
The block brush is an add-in, so you'll need to install it through the add-in manager window. 

Offhand, I don't know if it's still compatible with the latest development build though so there might be errors trying to install it or use it

**Comment by AshleyPikachu** [comment]
Alright, I learned how to disable it but it when I try to uninstall it properly it says its not found so its stuck. Thanks for letting me know.

**Comment by cameronwhite** [comment]
You might need to just clear out the addins cache folder. I think the path will depend on your system, but on Linux is likely a hidden folder in your home folder like `~/.mono-addins`

**Reply by AshleyPikachu** [reply]
I see, I currently uninstalled the development build of Pint  as its incredibly unstable on Windows & Mac. 

On Windows when I tried to copy part of an image and paste it into a new layer and it crashed without an error code.
On Mac I tried to expand the canvas while pasting the extract same image in, that did leave an error code.

Here is the source image and error, I hope this helps. Thanks for all the support.
![Pinta Mac Crash](https://github.com/PintaProject/Pinta/assets/26305479/90097bd0-7b83-4187-b976-cf8466a0ce4f)

https://wikimon.net/images/2/20/Kuwagamon.jpg

**Reply by cameronwhite** [reply]
Would you be able to expand the "Details" panel of that error dialog? I haven't been able to reproduce that crash while pasting

Also, do you recall which build you installed? I wonder if you had a old development build, since the Windows build in particular was very crash-prone until some recent fixes

**Reply by cameronwhite** [reply]
I think this crash was fixed by 257ecd7043a20e036778817d186994e67b04e3d2 so this was probably quite an old build
I'll update https://github.com/PintaProject/Pinta/releases/tag/dev-preview-builds if that's where you downloaded it from

**Reply by AshleyPikachu** [reply]
Yup, that's where I got it from. Its my sisters mac, so I will have the error code in a bit.

**Reply by AshleyPikachu** [reply]
Okay she says its not crashing on mac now.

**Reply by AshleyPikachu** [reply]
Hi, so apparently I'm having issues on windows even with pinta 2.2 being uninstalled and the github build being reinstalled. Here's what I found out. I hope its helpful.

System.ArgumentException: '0' cannot be greater than -520.
   at System.Math.ThrowMinMaxException[T](T min, T max)
   at Pinta.Actions.PasteAction.Paste(Document doc, Boolean toNewLayer, PointI pastePosition) in D:\a\Pinta\Pinta\Pinta\Actions\Edit\PasteAction.cs:line 133
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at GLib.Internal.MainLoopSynchronizationContext.<>c__DisplayClass1_0.<Post>b__0()
   at GLib.Internal.MainLoopSynchronizationContext.<>c__DisplayClass3_0.<ScheduleAction>b__0()


Windows 11 Original Canvas Size 80x80 tried to expand canvas with kuwagamon image
using pinta build dev Pinta-2.2-dev1576 - from december 19th 2023
https://github.com/PintaProject/Pinta/releases/tag/dev-preview-builds

**Reply by cameronwhite** [reply, marked as answer]
Could you try installing the latest build? I just updated https://github.com/PintaProject/Pinta/releases/tag/dev-preview-builds with build 1876 - that crash should have been fixed since the dev1576 build

**Reply by AshleyPikachu** [reply]
Yup that fixed it! Also, it took me forever but I did manage to manually uninstall the broken plugins. Thanks a ton for the help. If you need any more help testing I'm easiest to reach at the discord here.

https://discord.gg/XrDXVjNwdG

**Reply by cameronwhite** [reply]
great, thanks for confirming!
